bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/ExpireProfile.java

```diff
@@ -0,0 +1,233 @@
+        DEFAULT_STATES.add(HSBType.class);
+        DEFAULT_STATES.add(QuantityType.class);
+        DEFAULT_STATES.add(DecimalType.class);
+        DEFAULT_STATES.add(StringListType.class);
+        DEFAULT_STATES.add(StringType.class);
+        DEFAULT_STATES.add(RawType.class);
+    }
+
+    private final ProfileCallback callback;
+    private final ProfileContext context;
+    private @NonNullByDefault({}) ScheduledExecutorService executor;
+
+    /**
+     * Human readable textual representation of duration (e.g. "13h 42m 12s")
+     */
+    private @Nullable String durationString;
+
+    /**
+     * duration in ms
+     */
+    private long duration;
+
+    /**
+     * state to post if item is expired.
+     */
+    private State expireState = UnDefType.UNDEF;
+
+    /**
+     * command to post if item is expired.
+     */
+    private @Nullable Command expireCommand;
+
+    public ExpireProfile(ProfileCallback callback, ProfileContext context) {
+        this.callback = callback;
+        this.context = context;
+
+        Object durationValue = this.context.getConfiguration().get(DURATION_PARAM);
+        logger.debug("Configuring profile with {} parameter '{}'", DURATION_PARAM, durationValue);
+
+        if (!(durationValue instanceof String)) {
+            logger.error("Parameter '{}' is not of type String", DURATION_PARAM);
+            return;
+        }
+
+        durationString = (String) durationValue;
+        duration = parseDuration(durationString);
+        if (duration == -1) {
+            logger.error("Invalid duration: '{}' in parameter '{}'. Expected something like: '1h 15m 30s'", duration,
+                    DURATION_PARAM);
+            return;
+        }
+
+        Object stateValue = this.context.getConfiguration().get(STATE_PARAM);
+        Object commandValue = this.context.getConfiguration().get(COMMAND_PARAM);
+
+        if (stateValue instanceof String || commandValue instanceof String) {
+            if (commandValue != null) {
+                expireCommand = TypeParser.parseCommand(DEFAULT_COMMANDS, (String) commandValue);
+            } else {
+                expireState = TypeParser.parseState(DEFAULT_STATES, (String) stateValue);
+            }
+        }
+        executor = this.context.getExecutorService();
+    }
+
+    @Override
+    public ProfileTypeUID getProfileTypeUID() {
+        return SystemProfiles.EXPIRE;
+    }
+
+    @Override
+    public void onCommandFromItem(Command command) {
+        applyExpire(command);
+    }
+
+    @Override
+    public void onCommandFromHandler(Command command) {
+        applyExpire(command);
+    }
+
+    @Override
+    public void onStateUpdateFromItem(State state) {
+        applyExpire(state);
+    }
+
+    @Override
+    public void onStateUpdateFromHandler(State state) {
+        applyExpire(state);
+    }
+
+    private void applyExpire(Type state) {
+        logger.trace("Received '{}'", state);
+        executor.shutdownNow();
```

Comment:
Why do you shutdown the whole executor?
I assume you would like to remember the Future that is returned by `executor::schedule` and cancel that specific future.

Reply:
Do you feel this is really needed, because at one time there is only 0 or 1 Future scheduled, and when receiving a state it will be either cancelled, either rescheduled. So I felt this was fine.

Reply:
Perhaps I am wrong but my reading and understanding has been that there is one executor for all profiles.
See https://github.com/openhab/openhab-core/blob/48d873a/bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ProfileContextImpl.java#L45

A executor per profile will result into a lot of executors and overhead so the current implementation (using one executor for all profiles) makes sense (but that does not care as it is already the case).

So, if one profile shutdown the executor this will affect all the other profiles and stop all the scheduled jobs of every profile.